src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -33,184 +33,103 @@
 #include <stdint.h>
 #include <string>
 
+#include "application_manager/commands/mobile/add_sub_menu_request.h"
+
 #include "gtest/gtest.h"
-#include "utils/shared_ptr.h"
-#include "smart_objects/smart_object.h"
-#include "application_manager/smart_object_keys.h"
 #include "application_manager/commands/commands_test.h"
 #include "application_manager/commands/command_request_test.h"
-#include "application_manager/application.h"
 #include "application_manager/mock_application_manager.h"
 #include "application_manager/mock_application.h"
 #include "application_manager/mock_message_helper.h"
 #include "application_manager/event_engine/event.h"
-#include "application_manager/commands/mobile/add_sub_menu_request.h"
-#include "interfaces/MOBILE_API.h"
-#include "application_manager/policies/policy_handler_interface.h"
+#include "application_manager/mock_hmi_interface.h"
 
 namespace test {
 namespace components {
 namespace commands_test {
 namespace mobile_commands_test {
 
-using ::testing::_;
-using ::testing::Return;
-using ::testing::ReturnRef;
 namespace am = ::application_manager;
 using am::commands::AddSubMenuRequest;
 using am::commands::MessageSharedPtr;
 using am::event_engine::Event;
+using am::MockHmiInterfaces;
+using am::MockMessageHelper;
+using ::testing::_;
+using ::testing::Mock;
+using ::testing::Return;
 
 typedef SharedPtr<AddSubMenuRequest> AddSubMenuPtr;
 
 namespace {
 const uint32_t kConnectionKey = 2u;
-const int32_t kMenuId = 5;
 }  // namespace
 
 class AddSubMenuRequestTest
-    : public CommandRequestTest<CommandsTestMocks::kIsNice> {};
-
-TEST_F(AddSubMenuRequestTest, OnEvent_UnknownEvent_UNSUCCESS) {
-  Event event(hmi_apis::FunctionID::INVALID_ENUM);
-  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>());
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  AddSubMenuRequestTest()
+      : mock_message_helper_(*MockMessageHelper::message_helper_mock()) {}
```

Comment:
@okozlovlux, please, verify and clear expectations before use `mock_message_helper_`